Remotion/Data/DomainObjects.UnitTests/Queries/Configuration/QueryConfigurationTest.cs

```diff
@@ -399,5 +380,22 @@ public void Load_ProviderFromUndefinedStorageGroup ()
           queries["QueryFromUndefinedStorageGroup"].StorageProviderDefinition,
           Is.SameAs(DomainObjectsConfiguration.Current.Storage.DefaultStorageProviderDefinition));
     }
+
+#if NETFRAMEWORK
+    [Test]
+    public void CodeRunInSeparateAppDomain_AppDomainSetupSetsPropertiesCorrectly ()
+    {
+      AppDomainSetup setup = AppDomain.CurrentDomain.SetupInformation;
+      setup.ApplicationBase = @"C:\Base\";
+      setup.DynamicBase = Path.GetTempPath ();
+      setup.PrivateBinPath = @"Custom\Relative;Search\Path";
+
+      new AppDomainRunner (setup, delegate
+      {
+        Assert.That (AppDomain.CurrentDomain.BaseDirectory, Is.EqualTo (@"C:\Base\"));
+        Assert.That (AppDomain.CurrentDomain.RelativeSearchPath, Is.EqualTo (@"Custom\Relative;Search\Path"));
```

Comment:
Assert against `IAppContextProvider` implementation instead of AppDomain.
(1 test for `NETFRAMEWORK`, 1 for `!NETFRAMEWORK`)